openlibrary/plugins/openlibrary/js/carousels.js

```diff
@@ -59,6 +61,65 @@ const Carousel = {
             slidesToScroll: a,
             responsive: responsive_settings
         });
+
+        var availabilityStatuses = {
+            'open': {cls: 'cta-btn--available', cta: 'Read'},
+            'borrow_available': {cls: 'cta-btn--available', cta: 'Borrow'},
+            'borrow_unavailable': {cls: 'cta-btn--unavailable', cta: 'Join Waitlist'},
+            'error': {cls: 'cta-btn--missing', cta: 'No eBook'}
+        };
+
+        var addWork = function(work) {
+            var availability = work.availability.status;
+            var ocaid = work.availability.identifier;
+            var cls = availabilityStatuses[availability].cls;
+            var url = (cls == 'cta-btn--available') ? ('/borrow/ia/' + ocaid) : 
+                (cls == 'cta-btn--unavailable') ? ('/books/' + work.availability.openlibrary_edition) :
+                work.key;
+            var cta = availabilityStatuses[availability].cta;
+            console.log(cta);
+            var isClickable = availability == 'error' ? 'disabled' : '';
+
+            return '<div class="book carousel__item slick-slide slick-active" ' +
```

Comment:
It's been a while since I've been in the code base. Do we have a template library we're using? If not, I'll add it to my list of topics to discuss while I'm in town.

Reply:
Not yet but we could! I'm not sure how much we do this in JS right now but template strings would be a good interim solution.

Reply:
+1